src/controllers/auth.js

```diff
@@ -11,15 +10,16 @@ exports.signup = async (req, res, next) => {
   const { email, password, username } = req.body;
 
   if (req.params.type === 'volunteer') {
-    const volunteer = await Volunteer.findOne({ email });
+    const volunteer = await User.findOne({ email });
     if (volunteer) {
       return next(new ErrorResponse('User already exists', 400));
     }
     const hashedPassword = await bcrypt.hash(password, saltRounds);
-    const user = new Volunteer({
+    const user = new User({
       email,
       password: hashedPassword,
       username,
```

Comment:
Great usage of the new model. I wonder if we can create just one new User instead of implementing two separate ones. Like this below:
```
const user = new User({
      email,
      password: hashedPassword,
      username,
      type: req.params.type
    });
```